src/Core/Extensible.php

```diff
@@ -18,7 +17,18 @@
  */
 trait Extensible
 {
-    use CustomMethods;
+    use CustomMethods {
```

Comment:
I was suggesting that you edit CustomMethods::call:

```php
        switch (true) {
            case isset($config['property']) : {
+                if (!empty($config['warmPropertyMethod'])) {
+                   if (!method_exists($this, $method)) {
+                      $methodName = $config['warmPropertyMethod'];
+                      $this->$methodName();
+                   }
+                }

                $obj = $config['index'] !== null ?
                    $this->{$config['property']}[$config['index']] :
                    $this->{$config['property']};

                if ($obj) {
                    if (!empty($config['callSetOwnerFirst'])) {
                        /** @var Extension $obj */
                        $obj->setOwner($this);
                    }
                    $retVal = call_user_func_array(array($obj, $method), $arguments);
                    if (!empty($config['callSetOwnerFirst'])) {
                        /** @var Extension $obj */
                        $obj->clearOwner();
                    }
                    return $retVal;
                }
```

And then set warmPropertyMethod in CustomMethod::addMethodsFrom() the same way that callSetOwnerFirst is set.

It's not as loosely coupled but reducing the method-call count would be good in such a critical loop